snap/snapcraft.yaml

```diff
@@ -0,0 +1,117 @@
+name: synfigstudio
+base: core18
+adopt-info: synfig-studio
+confinement: strict
+license: GPL-3.0
+
+apps:
+  synfigstudio: &app
+    command: usr/bin/synfigstudio
+    common-id: org.synfig.SynfigStudio
+    desktop: usr/share/applications/org.synfig.SynfigStudio.desktop
+    extensions: [gnome-3-34]
+    plugs: &plugs [opengl, home, removable-media, gsettings, network]
+    environment: &env
+      __EGL_VENDOR_LIBRARY_DIRS: $SNAP/gnome-platform/usr/share/glvnd/egl_vendor.d:$SNAP/usr/share/glvnd/egl_vendor.d
+  
+  synfig: 
+    command: usr/bin/synfig
+    extensions: [gnome-3-34]
+    plugs: *plugs
+    environment: *env
+
+layout:
+  /usr/share/synfig: 
+    symlink: $SNAP/usr/share/synfig
+  /usr/share/pixmaps/synfigstudio:
+    symlink: $SNAP/usr/share/pixmaps/synfigstudio
+
+parts:
+  synfig-deps:
+    plugin: nil
+    build-packages:
+      - g++
+      - intltool
+      - gettext
+      - libmagickcore-dev
+      - libmagick++-dev
+      - libopenexr-dev
+      - libsigc++-2.0-dev
+      - libxml++2.6-dev
+      - libfftw3-dev
+      - libmlt-dev
+      - libmlt++-dev
+      - libmlt-data
+      - libboost-system-dev
+      - libmng-dev
+      - git
+    stage-packages:
+      - libilmbase12
+      - libmagick++-6.q16-7
+      - libmagic1
+      - libmlt++3
+      - libmlt6
+      - libmng2
+      - libopenexr22
+      - libsigc++-2.0-0v5
+      - libxml++2.6-2v5
+      - libdv-bin # mod_dv
+      - ffmpeg # mod_ffmpeg
+      - imagemagick # mod_imagemagick
+      - libslang2
+      - libglu1-mesa
+      - freeglut3
+
+  etl:
+    after: [synfig-deps]
+    plugin: autotools
+    source: ETL
+    configflags: &cfgflags
+      - --prefix=/usr
+
+  synfig-core:
+    after: [etl]
+    plugin: autotools
+    source: synfig-core
+    configflags: *cfgflags
+    override-build: &build |
+      set -xe
+      autoreconf -i
+      intltoolize --force
+      snapcraftctl build
+
+  synfig-studio:
+    after: [synfig-core]
+    plugin: autotools
+    source: synfig-studio
+    configflags: *cfgflags
+    override-build: *build
+    parse-info: [usr/share/appdata/org.synfig.SynfigStudio.appdata.xml]
```

Comment:
We pull description + version from the appdata file. Maybe we want to set the version according to git instead? E.g. via git describe or by appending the commit hash.

Reply:
Atm, each revision in the snap store would be versioned "1.5.0". It's a bit confusing for the user. New revision, but the version stays the same. We could do something like "1.5.0~48b6b5d".

Reply:
I think for nightly builds we need a version with commit hash and date of commit. Like this - 1.5.0-20200514_48b6b5d.
In this case the builds can be sorted.

For releases (development or stable) it is enough to have just version number - i.e. 1.5.0.

All stable/development releases have a corresponding tag on them (i.e. v1.2.0, v1.2.1, v1.3.1, etc). Maybe this can be used to distinguish nightly builds from releases (if needed).

Reply:
Okay. So we do it like this:

```
version = read from appdata file
if commit is not tagged:
  version = version + date + hash
```

Reply:
The parse-info stuff clashes with setting the version manually, so I have pulled the appdata summary and description into snapcraft.yaml and only get the version from there.